Assets/Scripts/GameState/PauseManager.cs

```diff
@@ -42,9 +45,23 @@ void Start()
 
     public void PauseGame()
     {
-        IsPaused = !IsPaused;
-        Time.timeScale = IsPaused ? 0 : initialTimeScale;
-        ui.SetActive(IsPaused);
+        if (gameObject.activeSelf)
+        {
+            IsPaused = !IsPaused;
+            Time.timeScale = IsPaused ? 0 : initialTimeScale;
+            ui.SetActive(IsPaused);
+
+            // Pause every audiosource in array.
+            foreach (AudioSource a in audioSources)
+            {
+                if (IsPaused)
+                    a.Pause();
+                else
+                    a.UnPause();
+            }
+            // Reduce the listener volume level
+            AudioListener.volume = IsPaused ? 0.3f : 1f;
```

Comment:
These two "magic numbers" should preferably be extracted to serialized fields, both so that they're named, and so that they can be tweaked through the Unity inspector 🙂
```suggestion
            AudioListener.volume = IsPaused ? pauseAudioVolume : normalAudioVolume;
```
(or choose some other variable names :))